soccer/src/soccer/coach/coach_node.cpp

```diff
@@ -0,0 +1,76 @@
+#include "coach_node.hpp"
+
+using std::placeholders::_1;
+
+CoachNode::CoachNode(const rclcpp::NodeOptions& options) : Node("coach_node", options) {
+    coach_pub_ = this->create_publisher<rj_msgs::msg::Coach>("/strategy/coach", 10);
+    playstate_change_timer_ = this->create_wall_timer(
+        100ms, std::bind(&CoachNode::check_for_playstate_change, this));
+    playstate_sub_ = this->create_subscription<rj_msgs::msg::PlayState>(
+        "/referee/play_state", 10, std::bind(&CoachNode::playstate_callback, this, _1));
+
+    current_play_state_.state = PlayState::State::Halt;
+    current_play_state_.restart = PlayState::Restart::Kickoff;
+    current_play_state_.our_restart = true;
+
+    rj_geometry_msgs::msg::Point temp_point;
+    temp_point.x = -1;
+    temp_point.y = -1;
+    current_play_state_.placement_point = temp_point;
+}
+
+void CoachNode::playstate_callback(rj_msgs::msg::PlayState::SharedPtr msg) {
```

Comment:
do what the warning suggests ([context](https://stackoverflow.com/questions/2627166/what-is-the-difference-between-a-const-reference-and-normal-parameter))

pretty sure it should look like this, but double-check with compiler
```suggestion
void CoachNode::playstate_callback(const PlayState::SharedPtr &msg) {
```